src/models/Token.ts

```diff
@@ -0,0 +1,41 @@
+import { getContract } from '@/config/contracts';
+import { getContractFromAbi } from '@/util/network';
+import mongoose from 'mongoose';
+import { Contract } from 'web3-eth-contract';
+
+export interface Token {
+    name: string;
+    symbol: string;
+    totalSuplly: string;
+    blockNumber: number;
+    transactionHash: string;
+    contract: Contract;
+    network: number;
+    sub: string;
+}
+
+export type TokenDocument = mongoose.Document & Token;
+
+const tokenSchema = new mongoose.Schema(
```

Comment:
@GarfDev This needs renaming to ERC20 since that is functionally correct and aligns better with the introduction of ERC721 (which will get its own collection, since it's properties will differ too much).

Reply:
@GarfDev  I'm also missing the `enum ERC20Type {0 = LimitedSupply; 1 = UnlimitedSupply; }` enum and storage for that value per ERC20 token.